src/lib/es2020.intl.d.ts

```diff
@@ -406,6 +401,31 @@ declare namespace Intl {
          *
          * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/supportedLocalesOf).
          */
-        supportedLocalesOf(locales?: LocalesArgument, options?: { localeMatcher?: RelativeTimeFormatLocaleMatcher; }): BCP47LanguageTag[];
+        supportedLocalesOf(locales?: LocalesArgument, options?: { localeMatcher?: RelativeTimeFormatLocaleMatcher; }): UnicodeBCP47LocaleIdentifier[];
     };
+
+    interface CollatorConstructor {
+        new (locales?: LocalesArgument, options?: CollatorOptions): Collator;
+        (locales?: LocalesArgument, options?: CollatorOptions): Collator;
+        supportedLocalesOf(locales: LocalesArgument, options?: CollatorOptions): string[];
+    }
+
+    interface DateTimeFormatConstructor {
+        new (locales?: LocalesArgument, options?: DateTimeFormatOptions): DateTimeFormat;
+        (locales?: LocalesArgument, options?: DateTimeFormatOptions): DateTimeFormat;
+        supportedLocalesOf(locales: LocalesArgument, options?: DateTimeFormatOptions): string[];
+    }
+
+    interface NumberFormatConstructor {
+        new (locales?: LocalesArgument, options?: NumberFormatOptions): NumberFormat;
+        (locales?: LocalesArgument, options?: NumberFormatOptions): NumberFormat;
+        supportedLocalesOf(locales: LocalesArgument, options?: NumberFormatOptions): string[];
+    }
+
+    interface PluralRulesConstructor {
+        new (locales?: LocalesArgument, options?: PluralRulesOptions): PluralRules;
+        (locales?: LocalesArgument, options?: PluralRulesOptions): PluralRules;

```

Comment:
is this needed? it's in es2018.intl.d.ts, so the constructors should merge from there if both are loaded.

I guess it's possible to specify es2020.intl without es2018.intl -- but in that case, there's no `const PluralRules` declared in es2020, so it's not usable anyway.

Edit: same question about all these constructors, except for es5 instead of es2018.intl

Reply:
Thanks for the explanation.
I missed the difference in the type of the `locales` parameters between the es2018 and es2020 versions. Since they're different, it does make sense for es2020 to add another overload.